src/Microsoft.AspNet.Mvc.Razor/RazorView.cs

```diff
@@ -93,22 +95,30 @@ public string Path
                 throw new ArgumentNullException(nameof(context));
             }
 
-            var bodyWriter = await RenderPageAsync(RazorPage, context, ViewStartPages);
-            await RenderLayoutAsync(context, bodyWriter);
+            var oldWriter = context.Writer;
+            var razorTextWriter = new RazorTextWriter(context.Writer, _htmlEncoder);
```

Comment:
We have to keep this around for things that continue writing to `ViewContext.Output` (like `Html.BeginForm`)